File: 1422_maximum_score_after_splitting_a_string/solution.rb
# LeetCode 1422 - Maximum Score After Splitting A String
# https://leetcode.com/problems/maximum-score-after-splitting-a-string/

def max_score(s)
  ones = s.count('1')
  left_zeros = answer = 0
  s[0...-1].each_char do |char|
    if char == '0'
      left_zeros += 1
    else
      ones -= 1
    end
    answer = [answer, left_zeros + ones].max
  end
  answer
end

File: 1422_maximum_score_after_splitting_a_string/Solution.swift
// LeetCode 1422 - Maximum Score After Splitting a String
// https://leetcode.com/problems/maximum-score-after-splitting-a-string/

class Solution {
    func maxScore(_ s: String) -> Int {
        var ones = s.filter { $0 == "1" }.count
        var leftZeros = 0, answer = 0
        let chars = Array(s)
        for char in chars.dropLast() {
            if char == "0" { leftZeros += 1 } else { ones -= 1 }
            answer = max(answer, leftZeros + ones)
        }
        return answer
    }
}
